src/map/pet.c

```diff
@@ -226,60 +274,66 @@ static int pet_sc_check(struct map_session_data *sd, int type)
 	return 0;
 }
 
+/**
+ * Updates a pet's hunger value and timer and updates the pet's intimacy value based on its new hunger value.
+ *
+ * @param tid The pet's current hunger timer value.
+ * @param tick The amount of ticks to add to the pet's hunger timer.
+ * @param id The pet's master's account ID.
+ * @param data Unused.
+ * @return 1 on failure, 0 on success.
+ *
+ */
 static int pet_hungry(int tid, int64 tick, int id, intptr_t data)
 {
+	int interval;
 	struct map_session_data *sd;
 	struct pet_data *pd;
-	int interval;
 
-	sd=map->id2sd(id);
-	if(!sd)
+	if ((sd = map->id2sd(id)) == NULL)
 		return 1;
 
-	if(!sd->status.pet_id || !sd->pd)
+	if (sd->status.pet_id == 0 || (pd = sd->pd) == NULL)
 		return 1;
 
-	pd = sd->pd;
-	if(pd->pet_hungry_timer != tid){
-		ShowError("pet_hungry_timer %d != %d\n",pd->pet_hungry_timer,tid);
-		return 0;
+	if (pd->pet_hungry_timer != tid) {
+		ShowError("pet_hungry_timer %d != %d\n", pd->pet_hungry_timer, tid);
+		return 1;
 	}
+
 	pd->pet_hungry_timer = INVALID_TIMER;
 
-	if (pd->pet.intimate <= 0)
-		return 1; //You lost the pet already, the rest is irrelevant.
+	if (pd->pet.intimate <= PET_INTIMACY_NONE)
+		return 1; /// You lost the pet already, the rest is irrelevant.
+
+	if (battle_config.pet_hungry_delay_rate != 100)
+		interval = pd->petDB->hungry_delay * battle_config.pet_hungry_delay_rate / 100;
+	else
+		interval = pd->petDB->hungry_delay;
 
 	pd->pet.hungry--;
-	/* Pet Autofeed */
+
+	/** Pet Autofeed */
```

Comment:
if it starts with a single * let it end up with single *, if two then let it end up with two etc..